backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java

```diff
@@ -211,6 +226,37 @@ public CommitDAORdbImpl(AuthService authService, RoleService roleService) {
               .setIdType(IDTypeEnum.IDType.VERSIONING_REPO_COMMIT_BLOB)
               .build(),
           datasetVersion.getTagsList());
+
+      long version = datasetVersion.getVersion();
```

Comment:
if the user sets a version we should use it.

Reply:
I can not set a version from the request, was that flow verified.

Reply:
fixed